anthony/unrealized-potential-of-string-diagrams.md

```diff
@@ -0,0 +1,128 @@
+---
+title: The Unrealized Potential of String Diagrams for ZKVMs
+publish_date: ???
+category: compilers
+image: media/string_puppet.webp
+excerpt: This post will describe string diagrams and explain why they are a well suited for designing ZKVMs.
+---
+
+## Trees Considered Harmful
+
+By the end of this post, I want to convince the reader that there are better forms of syntax than abstract syntax trees; namely, graphical ones. I'll also touch on how this may influence better ZKVM designs.
```

Comment:
Well, string diagrams in the form of proof nets date to the late-80s in the work of Jean-Yves Girard. This lead to interaction nets in the 90s work of Lafont. And the original inspiration was Penrose's work on graphical notations for tensor calculus from the early 1970s. There's also a genealogical connection to Feynman diagrams and the "Birdtracks" formalism for group theory. Personally, I don't find the history that interesting, and it would require litigating a long history of graphical notations based on poorly specified families of combinatorial objects. It's too much of a digression for a blog post that's not actually about history. Selinger's survey paper is a good source to cite, though. I usually give it when talking about string diagrams, but it didn't come to mind, this time.

Reply:
Right, I now better understand. This is at the level of generality of combinatorial objects. Wow, that is quite wide.